components/brave_wallet_ui/common/api/hardware_keyrings.ts

```diff
@@ -70,6 +77,13 @@ export function getLedgerFilecoinHardwareKeyring (): FilecoinLedgerKeyring {
   return filecoinHardwareKeyring
 }
 
+export function getLedgerSolanaHardwareKeyring (): SolanaLedgerBridgeKeyring {
```

Comment:
Can this function be abstracted in a way that's not specific to the coin type here as well like Trezor or does it behave differently in a way that doesn't work like that?

Reply:
I think our Trezor hardware wallet support may just be limited to ETH, so it might be giving the impression of working generally when it's just one case. Either way I don't think we can do this because the APIs we get to the ledger device provided by the ledger js deps vary depending on the chain (see [eth](https://github.com/LedgerHQ/ledgerjs/tree/master/packages/hw-app-eth) vs. [sol](https://github.com/LedgerHQ/ledgerjs/tree/master/packages/hw-app-solana)).

We do abstract a little bit a few lines above with getHardwareKeyring which accepts a coin type.

Reply:
👍🏻 Yeah I looked a bit more into this after I wrote this comment and I'm good with keeping this as is. Turns out there's been an [open PR ](https://github.com/trezor/trezor-firmware/pull/1521) for trezor to add support for solana for awhile and it's still not been merged